### PR TITLE
Implement request #986 rename systems.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This PR add the ability to rename systems as per request #986.

It follows #993.  After #993 is merged it is only 4 commits.

It add a context menu element to the SidePanel system name to rename the system if the system is owned by the player.

This PR changes the game mechanic of system ownership.  Previously systems could not be owned.  This PR changes system ownership.   An empire owns a system if is owns at least one planet in a system and no other empire owns any planets.  A system is unowned if no empire owns any planets or multiple empires own planets.

### CODE REVIEW

**Comment by geoffthemedio** [comment]
Can you do this without putting in all the unrelated Font reworking commits?

**Comment by LGM-Doyle** [comment]
Yes, if I want to do it now and then redo it later.
This was easier, because it sits on top of the `SidePanel` code which uses the list box changes.

**Comment by adrianbroher** [comment]
Sneaking in unrelated changes is not welcome.

See also https://github.com/freeorion/freeorion/blob/master/CONTRIBUTING.md#i-want-to-contribute-code

**Comment by LGM-Doyle** [comment]
What was I sneaking in?

**Comment by adrianbroher** [comment]
> What was I sneaking in?

#989 and #993

**Comment by LGM-Doyle** [comment]
I thought that I had clearly stated in the PR that it followed #993.  The intent was so that no one would waste time implementing #986 when it was already done.

In future how would you suggest I approach this if explaining it in the PR is in adequate?

**Comment by adrianbroher** [comment]
> In future how would you suggest I approach this if explaining it in the PR is in adequate?

The problem is not that you noted it in the OP.  The problem I have with this is that this is putting up pressure to accept the PRs without review by making an unrelated and somewhat popular feature depend on it.

Currently it is not even clear if the mentioned PRs will be accepted regardless of the well intention and thought put into it because nobody dares to really review it (see https://github.com/freeorion/freeorion/pull/989#pullrequestreview-2006282 and https://github.com/freeorion/freeorion/pull/989#issuecomment-249640103).

**Comment by LGM-Doyle** [comment]
That is fair.

I didn't intend to pressure anyone to accept the prior commits.  I just didn't feel like writing the code twice.

**Comment by adrianbroher** [comment]
> I just didn't feel like writing the code twice.

I can understand that perfectly.  But unless the depending PRs are merged there is nothing to do.

Reopening the PR because the response wasn't well thought out.

**Comment by LGM-Doyle** [comment]
Thanks.

If the other PRs are not merged, I'll refactor it to work on its own.

**Comment by geoffthemedio** [comment]
rebase?

**Comment by LGM-Doyle** [comment]
Rebased.

File: universe/System.h [hunk 1]
```diff
@@ -48,6 +48,10 @@ class FO_COMMON_API System : public UniverseObject {
 
     virtual int             SystemID() const                { return this->ID(); }
 
+    /** Systems are unowned unless at least one planet is owned by an empire and no other empire
+        owns any planets in the system.*/
+    int                     Owner() const;
```
Comment (geoffthemedio) [inline universe/System.h:53]: if the base class function is virtual, label the derived class as such as well

**Review by geoffthemedio — CHANGES_REQUESTED** [review]

**Pull request merged**